### PR TITLE
Add publications section to website

### PR DESCRIPTION
**Pull request by titu1994**
# What does this PR do ?

Adds a new publications section to the NeMo website

**Collection**: [Docs, Core, ASR, TTS, NLP]

# Changelog 
- Add a pseudo "blog" to incorporate all our publications
- Update some internal overrides to support useful redirects
- Make the `Continue reading` button have a right arrow plus be surrounded by a box to ensure users note where to click to go to the next page

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,6 +102,18 @@ up during the PR it must be a branch directly from NVIDIA/NeMo.
    - `OSS Community`: Contains the text regarding the NeMo integration with PTL and Hydra.
    - `RIVA`: Contains the text regarding NVIDIA RIVA.
 
+# Steps to add a publication
+
+Most of the instructions for adding a publication are similar to the instructions for adding a new post above. The differences are
```
Comment (erastorgueva-nv) [inline README.md:107]: 'The differences are:'

File: docs/overrides/blog-post.html [hunk 2]
```diff
@@ -25,20 +25,39 @@
...
 
-    <aside class="mdx-author">
 <!--        <p class="p-blog">-->
 <!--        <img src="https://avatars.githubusercontent.com/{{ gh_author }}" alt="@{{ gh_author }}">-->
 <!--        </p>-->
 <!--        <p class="p-blog">-->
-        <span> 
-        <b>{{ page.meta.author[loop.index0] }}</b> · <a href="https://github.com/{{ gh_author }}">@{{ gh_author }}</a>
+
+        <!-- If number of authors of less than 3, put on new lines -->
```
Comment (erastorgueva-nv) [inline docs/overrides/blog-post.html:37]: 'number of authors is less than 3' will be more readable

File: docs/publications/template.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# page_path: Relative path to the image from the website root (e.g. /assets/images/). If specified, the image at this path will be used for the link preview. It is unlikely you will need this parameter - you can probably use og_image instead.
+# description: Description of the post for Rich URL previews
+---
+
+# Notes
+
+- Add folder inside `blogs/posts` directory with year
+- Copy the contents of this template to the folder.
+- Edit the contents.
+- Add a `<!-- more -->` tag to the blog post to indicate where it should say 'Continue reading' in the blogpost preview.
```
Comment (erastorgueva-nv) [inline docs/publications/template.md:28]: IIUC if "continue_url" is set, then the text underneath the `<!-- more -->` tag will never be visible on the website. We should make that clear.

**Review by erastorgueva-nv — CHANGES_REQUESTED** [review]
Great work!
I would just make it clear in the `publications/template.md` and maybe also the `README.md` that if "continue_url" is set, then the text underneath the `<!-- more -->` tag will never be visible on the website.

**Comment by erastorgueva-nv** [comment]
Also, we don't seem to be using `author_gh_user` in the publications template - can we get rid of it? @titu1994

File: docs/publications/template.md (hunk 3 above)
Comment (erastorgueva-nv) [reply docs/publications/template.md:28]: Correction: if "continue_url" is set, AND a URL is specified in `"# [Article Title](Article URL)"`, then there will be no links to the full blog post on the website, although the full blog post will be accessible if a user knows the URL to look for. So the content under the 'more' tag won't be findable by clicking, but will be findable if someone looks for the exact blog post link.

**Comment by titu1994** [comment]
There's shared code path between the blog and publications, so we need to keep the gh auth users. It's ignored anyway, and people can still add to it if they want

File: docs/publications/template.md (hunk 3 above)
Comment (titu1994) [reply docs/publications/template.md:28]: Good point, added a note with this detail.

**Review by erastorgueva-nv — APPROVED** [review]
LGTM

**Pull request merged**